src/ui/game/userlist/user_cell.cpp

```diff
@@ -64,10 +65,24 @@ bool GlobedUserCell::init(const PlayerStore::Entry& entry, const PlayerAccountDa
         .store(nameBtn)
         .collect();
 
-    CCSprite* badgeIcon = util::ui::createBadgeIfSpecial(data.specialUserData);
-    if (badgeIcon) {
-        util::ui::rescaleToMatch(badgeIcon, util::ui::BADGE_SIZE);
-        usernameLayout->addChild(badgeIcon);
+    // CCSprite* badgeIcon = util::ui::createBadgeIfSpecial(data.specialUserData);
+    // if (badgeIcon) {
+    //     util::ui::rescaleToMatch(badgeIcon, util::ui::BADGE_SIZE);
+    //     usernameLayout->addChild(badgeIcon);
+    // }
+
+    // badge with s
+    if (data.specialUserData.roles) {
+        std::vector<std::string> badgeVector = RoleManager::get().createRoleArray(data.specialUserData.roles.value());
+        if (!badgeVector.empty()) {
+            for (std::string spr : badgeVector) {
+                auto badge = util::ui::createBadge(spr);
+                if (badge) {
+                    util::ui::rescaleToMatch(badge, util::ui::BADGE_SIZE);
+                    usernameLayout->addChild(badge);
+                }
+            }
+        }
```

Comment:
this code is repeated 3 times in your pr, consider making a method, for example `util::ui::createBadgesIn(const SpecialUserData&, CCNode*)` that would take all the roles and add them to the menu